### PR TITLE
issue-1834 set PostgreSQL fillfactor for updatable tables

### PR DESCRIPTION
**Pull request by punktilious**
Signed-off-by: Robin Arnold <robin.arnold@ibm.com>

### CODE REVIEW

File: fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/control/FhirSchemaGenerator.java [hunk 1]
```diff
@@ -789,7 +810,7 @@ public void addResourceChangeLog(PhysicalDataModel pdm) {
                 .setTablespace(fhirTablespace)
                 .addPrivileges(resourceTablePrivileges)
                 .enableAccessControl(this.sessionVariable)
-                .addWiths(addWiths()) // New Column for V0017
+                .addWiths(customWiths) // Does not require fillfactor tuning
```
Comment (prb112) [inline fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/control/FhirSchemaGenerator.java:813]: makes sense

**Review by prb112 — APPROVED** [review]
LGTM

Q: Is there any odd behavior we should let downstream users know about post? e.g. a rebalancing after the fillfactor change, or is it naturally going to happen over time

**Comment by punktilious** [comment]
> LGTM
> 
> Q: Is there any odd behavior we should let downstream users know about post? e.g. a rebalancing after the fillfactor change, or is it naturally going to happen over time

It'll happen naturally over time. The fillfactor will only affect newly allocated blocks or if a full vacuum is performed (which is a heavyweight operation and should be guided by a DBA). I think it's sufficient to mention in the release notes.

**Comment by prb112** [comment]
> > LGTM
> > Q: Is there any odd behavior we should let downstream users know about post? e.g. a rebalancing after the fillfactor change, or is it naturally going to happen over time
> 
> It'll happen naturally over time. The fillfactor will only affect newly allocated blocks or if a full vacuum is performed (which is a heavyweight operation and should be guided by a DBA). I think it's sufficient to mention in the release notes.

Great - That's what I was hoping.

**Pull request merged**